doc/code/templates.rst

```diff
@@ -0,0 +1,156 @@
+.. _templates:
+
+Templates
+=========
+
+**Module name:** :mod:`pennylane.templates`
+
+.. currentmodule:: pennylane.templates
+
+This module provides a growing library of templates of common quantum
+machine learning circuit architectures that can be used to easily build,
+evaluate, and train more complex quantum machine learning models. In the
+quantum machine learning literature, such architectures are commonly known as an
+**ansatz**.
+
+PennyLane conceptually distinguishes two types of templates, **layer architectures** and **input embeddings**:
+
+* Layer architectures, found in :mod:`pennylane.templates.layers`, define sequences of gates that are repeated
+  like the layers in a neural network. They usually contain only trainable parameters.
+
+* Embeddings, found in :mod:`pennylane.templates.embeddings`, encode input features into the quantum state of the
+  circuit. These embeddings can also depend on trainable parameters, in which case the embedding is learnable.
+
+Each trainable template has a dedicated function in :mod:`pennylane.templates.parameters` which generates a list of
+**randomly initialized** arrays for the trainable **parameters**. The entries of the list contain valid positional
+arguments for the template, allowing for the syntax ``MyTemplate(*par_list)``.
```

Comment:
Not at the moment, it returns a list which has to be unpacked explicitly. Would you prefer the return of the parameter functions to be like "return a, b, c"?

Reply:
The question was actually not about how things are returned, but how users can access what is returned. I'm ok with returning everything as a list or tuple for consistency. However, it would be good to clarify in the docs that the user can use this in two ways:

1. Keep all variables in a list and just dereference
```full_params = cvqnnlayer()
CVQNNLayer(*full_params)
```
or
2. Store all variables separately (maybe user does something with the params, or wants them to keep separate references to them). 
```
Rot_params, BS_params, Sqz_params, ... = cvqnnlayer()
Sqz_params = np.min(Sqz_params, 2.0)
CVQNNLayer(Rot_Params, BS_params, Sqz_params, ...)
```
Note that the second way (automatically indexing into a returned list, tuple when unpacking function return) is automatically supported by Python without you needing to change the code. But would be good to give an example of this ind ocs

Reply:
Oh of course! Added the two user option to template landing page.